Makefile

```diff
@@ -22,32 +22,38 @@ ENDPOINT_NAME = main.c
 SRC_SRCS = $(addprefix $(SOURCE_F)/, $(SRC_NAMES))
 ENDPOINT_SRC = $(addprefix $(SOURCE_F)/, $(ENDPOINT_NAME))
 
-OBJS = $(SRC_SRCS:.c=.o)
-ENDPOINT_OBJ = $(ENDPOINT_SRC:.c=.o)
+OBJ_F = build/
+OBJS = $(addprefix $(OBJ_F), $(SRC_NAMES:.c=.o))
+ENDPOINT_OBJ = $(OBJ_F)$(ENDPOINT_NAME:.c=.o)
 INCLUDES = -I . -I libft
 
 TEST_NAMES = main_test.c input_to_text_tree_test.c
 TEST_SRCS = $(addprefix $(TEST_F)/, $(TEST_NAMES))
-TEST_OBJS = $(TEST_SRCS:.c=.o)
+TEST_OBJS = $(addprefix $(OBJ_F), $(TEST_NAMES:.c=.o))
 TEST_FNAME = $(TEST_F)/test
 
-all: pre $(NAME)
+DIRS = $(COMMANDS_F) $(INPUT_TO_TEXT_TREE_MOCK_F)
+OBJ_DIRS = $(addprefix $(OBJ_F), $(DIRS))
+
+vpath %.c $(SOURCE_F) $(SOURCE_F)/$(COMMANDS_F) $(SOURCE_F)/$(INPUT_TO_TEXT_TREE_MOCK_F)
+
+all: pre $(OBJ_DIRS) $(NAME)
+
+$(OBJ_DIRS):
+	$(PREFIX)mkdir -p $(OBJ_DIRS)
```

Comment:
folders are not created on rules `test` and `tania` at least... please fix this in this PR, even if the rule `test` won't work currently